internal/testserver/datastore/postgres.go

```diff
@@ -19,12 +21,34 @@ import (
 	"github.com/authzed/spicedb/pkg/secrets"
 )
 
+const (
+	POSTGRES_TEST_USER            = "postgres"
+	POSTGRES_TEST_PASSWORD        = "secret"
+	POSTGRES_TEST_PORT            = "5432"
+	POSTGRES_TEST_MAX_CONNECTIONS = "500"
+	PGBOUNCER_TEST_PORT           = "6432"
+)
+
+var enablePgbouncer = flag.Bool("enablePgbouncer", false, "run pgbouncer in front of postgres datastore tests")
+
+func TestMain(m *testing.M) {
+	flag.Parse()
+}
+
+type container struct {
+	hostHostname      string
+	hostPort          string
+	containerHostname string
+	containerPort     string
+}
+
 type postgresTester struct {
-	conn            *pgx.Conn
-	hostname        string
-	port            string
-	creds           string
-	targetMigration string
+	container
+	hostConn             *pgx.Conn
+	creds                string
+	targetMigration      string
+	pgbouncerProxy       *container
+	useContainerHostname bool
```

Comment:
The original implementation would conditionally set `hostname` and `port` to the container values vs the host values depending on if a network bridge was provided. This was getting rather confusing for me to track when piling on pgbouncer. So I updated this struct to track both the container and host values. I then moved the conditional checks into new helper functions: `getHostnameAndPort`, `getHostHostnameAndPort`, and `getContainerHostnameAndPort`.

I was hoping to make things more clear, but I'm happy to refactor if others dislike it.